client/tests/integration/roles.rs

```diff
@@ -0,0 +1,92 @@
+#![allow(clippy::restriction)]
+
+use std::{str::FromStr as _, time::Duration};
+
+use eyre::{eyre, Result};
+use iroha_client::client::{self, Client};
+use iroha_core::{prelude::AllowAll, smartcontracts::permissions::ValidatorBuilder};
+use iroha_data_model::{permissions::Permissions, prelude::*};
+use iroha_permissions_validators::public_blockchain::transfer;
+use test_network::{Peer as TestPeer, *};
+use tokio::runtime::Runtime;
+
+#[test]
+fn add_role_to_limit_transfer_count() -> Result<()> {
+    const PERIOD_MS: u64 = 5000;
+    const COUNT: u32 = 2;
+
+    let rt = Runtime::test();
+    let (_peer, mut test_client) = rt.block_on(<TestPeer>::start_test_with_permissions(
+        ValidatorBuilder::new()
+            .with_recursive_validator(transfer::ExecutionCountFitsInLimit)
+            .all_should_succeed(),
+        AllowAll.into(),
+    ));
+    wait_for_genesis_committed(&vec![test_client.clone()], 0);
+
+    let alice_id = <Account as Identifiable>::Id::from_str("alice@wonderland")?;
+    let bob_id = <Account as Identifiable>::Id::from_str("bob@wonderland")?;
+    let rose_definition_id = <AssetDefinition as Identifiable>::Id::from_str("rose#wonderland")?;
+    let alice_rose_id =
+        <Asset as Identifiable>::Id::new(rose_definition_id.clone(), alice_id.clone());
+    let bob_rose_id = <Asset as Identifiable>::Id::new(rose_definition_id, bob_id.clone());
+    let role_id = <Role as Identifiable>::Id::from_str("non_privileged_user")?;
+    let rose_value = get_asset_value(&mut test_client, alice_rose_id.clone())?;
+    assert!(rose_value > COUNT + 1);
```

Comment:
I'd add a comment that Alice has roses from genesis. Orillion devs might not know that.